src/clj/game/core/costs.clj

```diff
@@ -34,12 +34,21 @@
   (<= 0 (- (get-in @state [side :click]) (value cost))))
 (defmethod handler :click
   [cost state side eid _card]
-  (let [a (:action eid)]
+  (let [a (:action eid)
+        idx (get-in eid [:source-info :ability-idx])
+        source-abilities (get-in eid [:source :abilities])
+        is-game-action? (when (and (= :ability (:source-type eid))
+                                   (number? idx)
+                                   (seq source-abilities))
+                          (:action (nth source-abilities idx {})))
+        source-type (get-in eid [:source :type])]
     (swap! state update-in [:stats side :lose :click] (fnil + 0) (value cost))
     (deduct state side [:click (value cost)])
     (wait-for (trigger-event-sync state side (make-eid state eid)
                                   (if (= side :corp) :corp-spent-click :runner-spent-click)
                                   {:action a
+                                   :is-game-action? is-game-action?
+                                   :source-type source-type
```

Comment:
Why not just put the source here? Then you can check with `(resource? (:source context))`

Reply:
with how often we duplicate eids, I thought it might bloat the gamestate. Happy to change it though.

Reply:
hmm maybe! i just don't want to get back into the habit of checking for a string when we mean a type. with the function, the impl can change to a keyword instead, or another type of object. "when would this happen though?" i've considered moving types to keywords lol, it's possible!

Reply:
hmm. What if we just strip the hell out of the card. Something like `(select-keys c [:type :title :cid])`, and just pass it as `stripped-card`?

Maybe that's worse from a maintenance perspective, I'm not sure.

Reply:
That's probably the right way to go. Then keys can be added as abilities require.